SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -104,49 +104,52 @@ public DBSPExpression visitInputRef(RexInputRef inputRef) {
+                return new DBSPIntervalMillisLiteral(literal, type, Objects.requireNonNull(
+                        literal.getValueAs(BigDecimal.class)).longValue());
+            else if (type.is(DBSPTypeTimestamp.class)) {
+                return new DBSPTimestampLiteral(literal, type,
+                        Objects.requireNonNull(literal.getValueAs(TimestampString.class)));
+            } else if (type.is(DBSPTypeDate.class)) {
+                return new DBSPDateLiteral(literal, type, Objects.requireNonNull(literal.getValueAs(DateString.class)));
+            } else if (type.is(DBSPTypeGeoPoint.class)) {
+                Point point = literal.getValueAs(Point.class);
+                Coordinate c = Objects.requireNonNull(point).getCoordinate();
+                return new DBSPGeoPointLiteral(literal,
+                        new DBSPDoubleLiteral(c.getOrdinate(0)),
+                        new DBSPDoubleLiteral(c.getOrdinate(1)));
             }
-        }
-        else if (type.is(DBSPTypeDouble.class))
-            return new DBSPDoubleLiteral(Objects.requireNonNull(literal.getValueAs(Double.class)));
-        else if (type.is(DBSPTypeFloat.class))
-            return new DBSPFloatLiteral(Objects.requireNonNull(literal.getValueAs(Float.class)));
-        else if (type.is(DBSPTypeString.class))
-            return new DBSPStringLiteral(Objects.requireNonNull(literal.getValueAs(String.class)));
-        else if (type.is(DBSPTypeBool.class))
-            return new DBSPBoolLiteral(Objects.requireNonNull(literal.getValueAs(Boolean.class)));
-        else if (type.is(DBSPTypeDecimal.class))
-            return new DBSPDecimalLiteral(
-                    literal, type, Objects.requireNonNull(literal.getValueAs(BigDecimal.class)));
-        else if (type.is(DBSPTypeKeyword.class))
-            return new DBSPKeywordLiteral(literal, Objects.requireNonNull(literal.getValue()).toString());
-        else if (type.is(DBSPTypeMillisInterval.class))
-            return new DBSPIntervalMillisLiteral(literal, type, Objects.requireNonNull(
-                    literal.getValueAs(BigDecimal.class)).longValue());
-        else if (type.is(DBSPTypeTimestamp.class)) {
-            return new DBSPTimestampLiteral(literal, type,
-                    Objects.requireNonNull(literal.getValueAs(TimestampString.class)));
-        } else if (type.is(DBSPTypeDate.class)) {
-            return new DBSPDateLiteral(literal, type, Objects.requireNonNull(literal.getValueAs(DateString.class)));
-        } else if (type.is(DBSPTypeGeoPoint.class)) {
-            Point point = literal.getValueAs(Point.class);
-            Coordinate c = Objects.requireNonNull(point).getCoordinate();
-            return new DBSPGeoPointLiteral(literal,
-                    new DBSPDoubleLiteral(c.getOrdinate(0)),
-                    new DBSPDoubleLiteral(c.getOrdinate(1)));
+        } catch (Throwable ex) {
+            throw new Unimplemented(literal, ex);
```

Comment:
We shouldn't ever have catch blocks for kitchen-sink exception types like `Throwable`, `Exception` and `RTE`. In this particular case, there's no guarantee here that it has anything to do with a feature being unimplemented either?

I assume we want this throwable to eventually become a `CompilerMessages.Error(Throwable e)` ?

Reply:
Yes, unimplemented becomes an Error. 
If I discover any exception here that does not mean "unimplemented" I will change this.